src/components/files/fileHandler.ts

```diff
@@ -0,0 +1,119 @@
+import path from 'path'
+import fs from 'fs'
+import util from 'util'
+import { pipeline } from 'stream'
+import { nanoid } from 'nanoid'
+import { getAppRootDir } from '../utilityFunctions'
+import * as config from '../../config.json'
+import DBConnect from '../databaseConnect'
+import createThumbnail from './createThumbnails'
+const sharp = require('sharp')
+
+export const { filesFolderName } = config
+export const filesPath = path.join(getAppRootDir(), filesFolderName)
+
+interface HttpQueryParameters {
+  [key: string]: string
+}
+
+export function createFilesFolder() {
+  try {
+    fs.mkdirSync(path.join(getAppRootDir(), filesFolderName))
+  } catch {
+    // Folder already exists
+  }
+  // Move generic thumbnails to files root
+  fs.readdir(path.join(getAppRootDir(), 'images', 'generic_file_thumbnails'), (_, files) => {
+    files.forEach((file) =>
+      fs.copyFile(
+        path.join(getAppRootDir(), 'images', 'generic_file_thumbnails', file),
+        path.join(getAppRootDir(), filesFolderName, file),
+        () => {}
+      )
+    )
+  })
+}
+
+export async function getFilePath(uid: string, thumbnail = false) {
+  const result = await DBConnect.getFileDownloadInfo(uid, thumbnail)
+  if (!result) throw new Error()
+  return result
+}
+
+const pump = util.promisify(pipeline)
+
+export async function saveFiles(data: any, queryParams: HttpQueryParameters) {
+  const filesInfo = []
+  try {
+    for await (const file of data) {
+      const ext = path.extname(file.filename)
+      const basename = path.basename(file.filename, ext)
+      const unique_id = nanoid()
+
+      // Process file data/url queries
+      const parameters = { ...queryParams }
+      ;['user_id', 'application_serial', 'application_response_id'].forEach((field) => {
+        const queryFieldValue = queryParams[field]
+        const bodyFieldValue = file.fields[field] ? file.fields[field].value : undefined
+        parameters[field] = queryFieldValue ? queryFieldValue : bodyFieldValue
+      })
+
+      const subfolder = parameters?.application_serial ? parameters.application_serial : ''
```

Comment:
should you also create subfolder path ? (that could be passed to createThumbnail)